### PR TITLE
[pr2eus_moveit] send angle-vector only once in :angle-vector-motion-plan

### PR DESCRIPTION
**Pull request by k-okada**
rewrited veresion of #304

### CODE REVIEW

File: pr2eus_moveit/euslisp/robot-moveit.l [hunk 1]
```diff
@@ -552,22 +552,28 @@
...
+       (mapcar #'(lambda (ac) ;; for all installed controllers
+                   (let (controller-methods controller-definition)
+                     (ros::ros-debug "for installed controller ~A..." ac)
+                     ;;  controller within target ctype
+                     (ros::ros-debug "        a part of ~A : ~A" ctype (not (null (member ac (gethash ctype controller-table)))))
+                     ;; retrive controller method from action name
+                     (setq controller-methods (remove nil (mapcar #'(lambda (m) (if (find (send ac :name) (send self m) :test #'(lambda (a b) (string= a (cdr (assoc :controller-action b))))) m)) (send controller-table :list-keys))))
+                     (setq controller-definition (find (send ac :name) (send self (car controller-methods)) :test #'(lambda (a b) (string= a (cdr (assoc :controller-action b))))))
+                     ;; contains joint which is not included in move-arm
+                     (ros::ros-debug "        overap joint name for current move-arm : ~A" (not (intersection (send traj :joint-names) (cdr (assoc :joint-names controller-definition)) :test #'string=)))
```
Comment (knorth55) [inline pr2eus_moveit/euslisp/robot-moveit.l]: `(send traj :joint_names)`

File: pr2eus_moveit/euslisp/robot-moveit.l [hunk 2]
```diff
@@ -552,22 +552,28 @@
...
+                     (when (and (member ac (gethash ctype controller-table))
+                                (not (intersection (send traj :joint-names) (cdr (assoc :joint-names controller-definition)) :test #'string=)))
```
Comment (knorth55) [inline pr2eus_moveit/euslisp/robot-moveit.l]: `(send traj :joint_names)`

File: pr2eus_moveit/euslisp/robot-moveit.l [hunk 3]
```diff
@@ -552,22 +552,28 @@
...
+                       (ros::ros-debug "      send to ~A" ac)
+                       (if (listp av)
+                           (send-message self robot-interface :angle-vector-sequence av total-time ac)
+                         (send-message self robot-interface :angle-vector av total-time ac))
```
Comment (knorth55) [inline pr2eus_moveit/euslisp/robot-moveit.l]: `ac` is `action-client`, not `controller-type`
```
29.irteusgl$ (send *ri* :angle-vector (send *baxter* :angle-vector) 3000 :rarm-head-controller 0 :move-arm :rarm)
[ INFO] [1499424427.203097301, 1366.223000000]: ;; Planned Trajectory Total Time0.068 [sec]
[ INFO] [1499424427.203335272, 1366.223000000]: ;; Scaled Trajectory Total Time   3.000 [sec]
[ INFO] [1499424427.203374888, 1366.223000000]: ;; generated 10 points for 3 sec using [right_arm] group
;; controller-type: #<controller-action-client #X6c9e780 /robot/head/head_action> not found(#f(-80.0 20.0 -25.0 -40.0 60.0 -20.0 80.0 0.0 5.0 -57.0 68.0 111.0 -38.0 59.0 0.0))
```

Comment (knorth55) [inline pr2eus_moveit/euslisp/robot-moveit.l]: with this warn, `head` does not move.

**Comment by k-okada** [comment]
Closed via #308

**Pull request closed without merging**